p2p/protocol/identify/id.go

```diff
@@ -194,9 +192,7 @@ func (ids *IDService) loop() {
 			}
 
 			if ids.Host.Network().Connectedness(rp) == network.Connected {
-				mes := &pb.Identify{}
-				ids.populateMessage(mes, rp, addReq.localConnAddr, addReq.remoteConnAddr)
-				ph = newPeerHandler(rp, ids, mes)
+				ph = newPeerHandler(rp, ids)
 				ph.start()
 				phs[rp] = ph
 				addReq.resp <- ph
```

Comment:
If the peer disconnects between the time when the other goroutine sent into this one and now, we will never respond on the resp channel and we'll have a goroutine leak.

@Stebalien

Reply:
Hm. You're right. We need to send _something_ back then reset the stream, I guess.

Reply:
Fixed in https://github.com/libp2p/go-libp2p/pull/942